ReactCommon/react/renderer/mounting/TelemetryController.cpp

```diff
@@ -28,24 +28,22 @@ bool TelemetryController::pullTransaction(
 
   auto transaction = std::move(*optional);
 
-  auto surfaceId = transaction.getSurfaceId();
-  auto number = transaction.getNumber();
   auto telemetry = transaction.getTelemetry();
   auto numberOfMutations = static_cast<int>(transaction.getMutations().size());
 
   mutex_.lock();
   auto compoundTelemetry = compoundTelemetry_;
   mutex_.unlock();
 
-  willMount({surfaceId, number, telemetry, compoundTelemetry});
+  willMount(transaction);
 
   telemetry.willMount();
-  doMount(transaction.getMutations());
+  doMount(transaction);
   telemetry.didMount();
 
   compoundTelemetry.incorporate(telemetry, numberOfMutations);
 
-  didMount({surfaceId, number, telemetry, compoundTelemetry});
```

Comment:
I started updating internal usages of this method, seems like `compoundTelemetry` is not available in the `transaction` .
I am not sure what's the best course of action here, maybe we could pass both transaction and telemetry as separate parameters or rollback deletion of `metadata` and just have it as a `{ transaction, compoundTelemetry }` type of object?

Reply:
How about passing SurfaceTelemetry as a second argument to all the observer callbacks? For me it'd seem more natural than adding additional struct to carry them both. If you'd prefer to have such struct then I'd consider either naming it differently than "*Metadata" as it'd seem unintuitive that "TransactionMetadata" actually carries "Transaction". Alternatively I can revert "Metadata" to its previous form and add only a reference to "mutations" to it as that's the only thing we need from transaction to fulfil out usecase.

Reply:
Yep, SurfaceTelemetry as a separate parameter sounds good to me :)